gammapy/astro/darkmatter/profiles.py

```diff
@@ -0,0 +1,250 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Dark matter profiles
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from ...utils.modeling import Parameter, ParameterList
+from ...spectrum.utils import integrate_spectrum
+import astropy.units as u
+import six
+import abc
+import numpy as np
+
+__all__ = [
+    'DMProfile',
+    'NFWProfile',
+    'EinastoProfile',
+    'IsothermalProfile',
+    'BurkertProfile',
+    'MooreProfile',
+]
+
+
+@six.add_metaclass(abc.ABCMeta)
+class DMProfile(object):
+    """DMProfile model base class.
+    """
+    LOCAL_DENSITY = 0.3 * u.GeV / (u.cm ** 3)
+    """Local dark matter density as given in refenrece 2"""
+    DISTANCE_GC = 8.33 * u.kpc
+    """Distance to the Galactic Center as given in reference 2"""
+
+    def __call__(self, radius):
+        """Call evaluate method of derived classes"""
+        kwargs = dict()
+        for par in self.parameters.parameters:
+            kwargs[par.name] = par.quantity
+
+        return self.evaluate(radius, **kwargs)
+
+    def scale_to_local_density(self):
+        """Scale to local density"""
+        scale = (self.LOCAL_DENSITY / self(self.DISTANCE_GC)).to('').value
+        self.parameters['rho_s'].value *= scale
+
+    def _eval_squared(self, radius):
+        """Helper function to return squared density"""
+        val = self(radius)
+        return val ** 2
+
+    def integral(self, rmin, rmax, **kwargs):
+        r"""Integrate squared dark matter profile numerically.
+
+        .. math::
+
+            F(r_{min}, r_{max}) = \int_{r_{min}}^{r_{max}}\rho(r)^2 dr
+
+
+        Parameters
+        ----------
+        rmin, rmax : `~astropy.units.Quantity`
+            Lower and upper bound of integration range.
+        **kwargs : dict
+            Keyword arguments passed to :func:`~gammapy.spectrum.integrate_spectrum`
+        """
+        integral = integrate_spectrum(self._eval_squared, rmin, rmax, **kwargs)
+        return integral.to('GeV2 / cm5')
+
+
+class NFWProfile(DMProfile):
+    r"""NFW Profile.
+
+    .. math::
+
+        \rho(r) = \rho_s \frac{r_s}{r}\left(1 + \frac{r}{r_s}\right)^{-2}
+
+    Parameters
+    ----------
+    r_s : `~astropy.units.Quantity`
+        Scale radius, :math:`r_s`
+    rho_s : `~astropy.units.Quantity`
+        Characteristic density, :math:`\rho_s`
+
+    References
+    ----------
+    * `arXiv:astro-ph/9611107 <https://arxiv.org/abs/astro-ph/9611107>`_
+    * `arXiv:1012.4515 <https://arxiv.org/abs/1012.451>`_
```

Comment:
"Paper identifier '1012.451' not recognized"
I think it's this one: http://adsabs.harvard.edu/abs/2011JCAP...03..051C